### PR TITLE
:seedling: Bump internal versions to use release v0.4.0

### PR DESCRIPTION
**Pull request by killianmuldoon**
Update the internally referenced version of the action and the notes tool to the current release.

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/hold

This is breaking the changes made for Cluster API in https://github.com/kubernetes-sigs/kubebuilder-release-tools/pull/55 as the version used in the hosted action does not include those changes.

AFAICT the version here should probably be bumped to v0.4.1 - is my understanding correct?

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
I would guess it should be v0.4.1 to match the upcoming release.

@vincepri @camilamacedo86?

**Review by camilamacedo86 — APPROVED** [review]
/lgtm

File: verify/go.mod
Comment (camilamacedo86) [reply verify/go.mod:10]: we can get this one merged and then do another one
or just set here the latest version.

**Comment by killianmuldoon** [comment]
@camilamacedo86 are you likely to do a new release of 0.4.0 for this? In that case I'll bump the version in this PR to v0.4.0

**Comment by sbueringer** [comment]
> @camilamacedo86 are you likely to do a new release of 0.4.X for this? In that case I'll bump the version in this PR to v0.4.1

@killianmuldoon Is your action taking the version from the master branch or from the tag? If it's from the tag I would prefer v0.4.1 & a new patch release to have it clean

**Comment by killianmuldoon** [comment]
>@killianmuldoon Is our action taking the version from the master branch or from the tag? If it's from the tag I would prefer v0.4.1 & a new patch release to have it clean

It's from the tag - but my question is to figure out the preference of whoever is doing the release. I'll update the PR regardless

**Comment by sbueringer** [comment]
I think it would be pretty bad if we basically consume a tag which points to another version (e.g. we pin to v0.4.1 and that links to v0.4.0).

But of course for this PR here it depends on if we just bring main in line with the last release (so set it to v0.4.0) or if we already set it up for a potential v0.4.1 release. I would prefer the latter, but as you said of course upt o maintainers.

**Comment by sbueringer** [comment]
/lgtm

from my side, but should be approved by a maintainer before merge

**Review by camilamacedo86 — COMMENTED** [review]
@vincepri is who has been taken care of this one.
I am OK with the changes, but let's see what he thinks.

**Comment by sbueringer** [comment]
@vincepri / @camilamacedo86 Can one of you please create a new v0.4.1 release?

We currently use v0.4.0 of the action (which uses the v0.3.0 image) in Cluster API and we would like to resolve this situation

**Pull request merged**
